ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -505,6 +507,14 @@ private void maybeWriteToChannel() {
     void completeRead() {
       if (readIntoCallbackCalled.compareAndSet(false, true)) {
         Exception e = operationException.get();
+        if (e instanceof RouterException) {
+          RouterException routerException = (RouterException) e;
+          if (routerException.getErrorCode() == RouterErrorCode.UnexpectedInternalError) {
```

Comment:
Is there a specific reason to convert to the RestServiceException here instead of inside of FrontendRestRequestService.submitResponse (line 407-410)?

Reply:
No particular reason, it's just that we are dealing with RouterException here, so it's better to be dealt in router.